src/map/script.cpp

```diff
@@ -19901,7 +19922,16 @@ BUILDIN_FUNC(instance_mapname)
  *------------------------------------------*/
 BUILDIN_FUNC(instance_id)
 {
-	script_pushint(st, script_instancegetid(st));
+	if (script_hasdata(st, 2)) {
+		enum instance_mode mode = static_cast<instance_mode>(script_getnum(st, 2));
```

Comment:
Please treat mode as int here and only do a static_cast after you validated it(check below).
This can cause undefined behavior and is a no-go.